database/pgx/pgx.go

```diff
@@ -394,37 +401,68 @@ func (p *Postgres) Version() (version int, dirty bool, err error) {
 }
 
 func (p *Postgres) Drop() (err error) {
-	// select all tables in current schema
-	query := `SELECT table_name FROM information_schema.tables WHERE table_schema=(SELECT current_schema()) AND table_type='BASE TABLE'`
-	tables, err := p.conn.QueryContext(context.Background(), query)
+	// drop all tables in current schema
+	err = p.dropElement(tableSQL)
+	if err != nil {
+		return err
+	}
+
+	// drop all custom types in current schema
+	err = p.dropElement(typeSQL)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Postgres) dropElement(element sqlElement) (err error) {
+	// select all of "element" in current schema
+	var query string
+	switch element {
+	case tableSQL:
+		query = `SELECT table_name FROM information_schema.tables WHERE table_schema=(SELECT current_schema()) AND table_type='BASE TABLE'`
+	case typeSQL:
+		query = `
+			SELECT t.typname as type
+			FROM pg_type t
+			LEFT JOIN pg_catalog.pg_namespace n ON n.oid = t.typnamespace
+			WHERE (t.typrelid = 0 OR (SELECT c.relkind = 'c' FROM pg_catalog.pg_class c WHERE c.oid = t.typrelid))
```

Comment:
Can you add comments for this query? It's not immediately clear to me what the filtering conditions are doing due to my lack of familiarity with the postgres system catalogs.

Docs:
* https://www.postgresql.org/docs/current/catalog-pg-type.html
    * Do `typelm` and `typarray` track references to the type?
* https://www.postgresql.org/docs/current/catalog-pg-namespace.html
* https://www.postgresql.org/docs/current/catalog-pg-class.html

Also, shouldn't
```sql
WHERE (t.typrelid = 0 OR (SELECT c.relkind = 'c' FROM pg_catalog.pg_class c WHERE c.oid = t.typrelid))
```
be
```sql
WHERE (t.typrelid = 0 OR (SELECT 1 FROM pg_catalog.pg_class c WHERE c.oid = t.typrelid AND c.relkind = 'c'))
```

Reply:
Thanks for the documentation I had some time to look into it a bit. I'm very new to postgresql so a few weeks before this PR I did not even know arrays where possible. I took this query from stackoverflow and adjusted it for the current schema and tested it.

> Also, shouldn't
> `WHERE (t.typrelid = 0 OR (SELECT c.relkind = 'c' FROM pg_catalog.pg_class c WHERE c.oid = t.typrelid))`
> be
> `WHERE (t.typrelid = 0 OR (SELECT 1 FROM pg_catalog.pg_class c WHERE c.oid = t.typrelid AND c.relkind = 'c'))`

No the query is correct. Your suggestion will result in a syntax error ( subquery needs to return a boolean). I don´t fully understand why this is needed because I'm unfamiliar with the administration of composite types. What this WHERE does is include all composite and non-composite types. Types can apparently also have a typerel of a table or index etc.

I will do my best to formulate a comment